src/libraries/System.IO.Compression/tests/ZipArchive/zip_UpdateTests.cs

```diff
@@ -337,5 +338,159 @@ public void UpdateUncompressedArchive()
+            {
+                zip.Comment = comment;
+                Assert.Equal(expectedComment, zip.Comment);
+            }
+            // Read (verify creation)
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Read, leaveOpen: true))
+            {
+                Assert.Equal(expectedComment, zip.Comment);
+            }
+            // Update
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Update, leaveOpen: true))
+            {
+                zip.Comment = updatedComment;
+                Assert.Equal(expectedUpdatedComment, zip.Comment);
+            }
+            // Read (verify update)
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Read))
+            {
+                Assert.Equal(expectedUpdatedComment, zip.Comment);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TestComments))]
+        public static void Update_ZipArchiveEntry_Comment(string? comment)
+        {
+            string expectedComment = GetExpectedComment(comment);
+            string updatedComment = "Updated comment " + comment;
+            string expectedUpdatedComment = GetExpectedComment(updatedComment);
+
+            var stream = new MemoryStream();
+            var testStream = new WrappedStream(stream, true, true, true, null);
+
+            // Create
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Create, leaveOpen: true))
+            {
+                ZipArchiveEntry entry = zip.CreateEntry("testfile.txt", CompressionLevel.NoCompression);
+
+                entry.Comment = comment;
+                Assert.Equal(expectedComment, entry.Comment);
+            }
+            // Read (verify creation)
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Read, leaveOpen: true))
+            {
+                foreach (ZipArchiveEntry entry in zip.Entries)
+                {
+                    Assert.Equal(expectedComment, entry.Comment);
+                }
+            }
+            // Update
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Update, leaveOpen: true))
+            {
+                foreach (ZipArchiveEntry entry in zip.Entries)
+                {
+                    entry.Comment = updatedComment;
+                    Assert.Equal(expectedUpdatedComment, entry.Comment);
+                }
+            }
+            // Read (verify update)
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Read))
+            {
+                foreach (ZipArchiveEntry entry in zip.Entries)
+                {
+                    Assert.Equal(expectedUpdatedComment, entry.Comment);
+                }
+            }
+        }
+
+        [Fact]
+        // General purpose bit flag must get the appropriate bit set if a file comment is unicode
+        public static void Update_ZipArchive_And_ZipArchiveEntry_UnicodeComment()
```

Comment:
@GrabYourPitchforks is this an appropriate way of testing a comment with unicode characters?

Reply:
You can just use `Encoding.UTF8` as part of this test. The code paths controlled by the _encoderShouldEmitUTF8Identifier_ parameter aren't exercised by the `ZipArchive` class's implementation. But otherwise, yeah, seems correct. :)

Reply:
Ok thanks. I assume you mean:

- Removing line 428.
- Passing `Encoding.UTF8` directly as the last argument in line 437.